src/robot/libraries/Collections.py

```diff
@@ -804,9 +916,14 @@ def _log_dictionary(self, dictionary):
         for key in self.get_dictionary_keys(dictionary):
             yield f'{key}: {dictionary[key]}'
 
-    def _keys_should_be_equal(self, dict1, dict2, msg, values):
+    def _keys_should_be_equal(self, dict1, dict2, msg, values, ignore_case):
         keys1 = self.get_dictionary_keys(dict1)
         keys2 = self.get_dictionary_keys(dict2)
+        if ignore_case:
+            keys1 = self._normalize(dict1)
+            keys2 = self._normalize(dict2)
+            dict1 = self._normalize(dict1)
+            dict2 = self._normalize(dict2)
```

Comment:
There shouldn't be need to update this helper method at all. I believe normalization can be done already earlier. Same with the similar helper methods below.